variables.tf

```
@@ -92,6 +92,12 @@ variable "ecs_subnet_ids" {
   type        = list(string)
 }
 
+variable "assign_public_ip" {
+  description = "Whether this instance should be accessible from the public internet. Default is false."
+  default     = false
+  type        = string
```

```suggestion
  type        = bool
```

I think @esacteksab is correct here, this should be modified to `bool`.